src/bridge.ts

```diff
@@ -12,13 +12,14 @@ import { getBinaryExtension } from './platform.js'
 const DENO_VERSION_FILE = 'version.txt'
 const DENO_VERSION_RANGE = '^1.20.3'
 
-type LifecycleHook = () => void | Promise<void>
+type OnBeforeDownloadHook = () => void | Promise<void>
+type OnAfterDownloadHook = (error: Error | null) => void | Promise<void>
```

Comment:
Now that we supply the actual error, we can also make `error` optional instead of `Error | null`

```suggestion
type OnAfterDownloadHook = (error?: Error) => void | Promise<void>
```

This ensures better backwards-compatibility, even though this might be not really relevant here as this is only used internally. But also makes the usage a little simpler, see below (or above :)).